edx_toggles/toggles/internal/waffle.py

```diff
@@ -56,13 +56,26 @@ def is_enabled(self, switch_name):
         """
         Returns and caches whether the given waffle switch is enabled.
         """
-        namespaced_switch_name = self._namespaced_name(switch_name)
-        value = self._cached_switches.get(namespaced_switch_name)
+        value = self.get_request_cache(switch_name)
```

Comment:
I feel pretty strongly that I want the namespacing code killed immediately, without making any other changes before it is gone.  Here is why:
* The namespacing, and having the caches in the namespacing, makes the code much more complicated then necessary. 
* Having complicated code is painful to review, because I have to be extremely careful that everything is correct.  I don't see the point in this for me or for you.  It would be so much better to just review the simplified code.
* This change is a perfect example.  I'm not sure if the unit tests caught this or would have, but `switch_name` and `namespaced_switch_name` are not the same, and you were now getting the value from the cache using the wrong one.  This could not happen if everything were just using `self.name`, which will be so much simpler to read and understand.
* Note that the cache is just a RequestCache, so we need no backward compatibility on the cache setup.  It just needs to work.
Please tell me this works for you. ;)

Reply:
Actually, this is one of those changes that convince me that we should *not* make all changes at the same time :stuck_out_tongue: 

Let me describe what happened here:

1. I removed the override methods from the namespace classes to make the namespace classes progressively useless. ("empty shells", or "thin wrappers")
2. In the process I discovered that the `WaffleSwitchNamespace.override_for_request` method was used outside of tests. See [here](https://github.com/edx/edx-platform/pull/25417/files#diff-2de640ef4f32619056b959292135e77259f2b8708cd5bb0a581a61ddee5ff2e1) and [here](https://github.com/edx/edx-platform/pull/25417/files#diff-e1e457890b73461641c5f17bf345f9e862642575d777e88267707100fab29904).
3. To preserve this behaviour I decided to implement a `WaffleSwitchNamespace.set_request_cache` method: the implementation of this method is correct, as the waffle switch name is transformed in a namespaced name in the function body. This implementation made the API much simpler to write. I also decided to make a corresponding `get_request_cache` method to make those two methods the only entrypoints to the request cache.

The edx-platform code is rather unwieldy (as I'm sure you know :wink:). The [edx-platform pr #25417](https://github.com/edx/edx-platform/pull/25417) that removes the `.override` methods already affects more than a hundred file. I do not feel comfortable getting rid of the namespace classes and their features all in one go, thus resulting in an even bigger PR.

Reply:
That being said, I can start working on getting rid of the namespace classes. But this is going to result in a PR which will be based on top of  [pr #25417](https://github.com/edx/edx-platform/pull/25417), which is itself based on top of [pr #25286](https://github.com/edx/edx-platform/pull/25286). Having three huge PRs stacked on top of one another makes me think of a shaky Jenga tower...

Reply:
If anyone sees this later, we discussed offline and found a simpler approach.